runtime/lua/vim/filetype.lua

```diff
@@ -4,7 +4,7 @@ local fn = vim.fn
 local M = {}
 
 --- @alias vim.filetype.mapfn fun(path:string,bufnr:integer, ...):string?, fun(b:integer)?
---- @alias vim.filetype.maptbl [string|vim.filetype.mapfn, {priority:integer}]
+--- @alias vim.filetype.maptbl { [1]: string|vim.filetype.mapfn, [2]: { priority: number? } }
```

Comment:
Why change this from the tuple syntax?

Reply:
Oh, maybe I misunderstanding what LuaLS means by "tuple". Is it indeed the same as table but with known set of indexes? So are these two annotations equivalent?

Reply:
Yeah, I've updated LuaLS (from 3.7.4 to 3.9.3) and now warnings now.